### PR TITLE
Adds PHPunit and formats for PHPCS

### PR DESCRIPTION
**Pull request by davisshaver**
* Adds basic unit tests.
* Adds PHPdocs for WP-Docs standard.

### CODE REVIEW

File: phpcs.ruleset.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<ruleset name="WordPress Coding Standards for Plugins">
+	<description>Generally-applicable sniffs for WordPress plugins</description>
+
+	<rule ref="WordPress-Core" />
```
Comment (timatron) [inline phpcs.ruleset.xml]: does this cover WordPress-VIP now?

File: tests/test-photonfill-unit-test-case.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			// @codingStandardsIgnoreEnd
+		}
+		$upload = wp_upload_bits( $filename, null, $contents );
+		$this->assertTrue( empty( $upload['error'] ) );
+		$type = '';
+		if ( ! empty( $upload['type'] ) ) {
+			$type = $upload['type'];
+		} else {
+			$mime = wp_check_filetype( $upload['file'] );
+			if ( $mime ) {
```
Comment (timatron) [inline tests/test-photonfill-unit-test-case.php]: should this be a ! empty?

File: tests/test-photonfill-unit-test-case.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+			array(
+				'post_title' => basename( $upload['file'] ),
+				'post_content' => 'Test Attachment',
+				'post_type' => 'attachment',
+				'post_parent' => $parent_post_id,
+				'post_mime_type' => $type,
+				'guid' => $upload['url'],
+			)
+		);
+		$id = wp_insert_attachment( $attachment, $upload['file'], $parent_post_id );
```
Comment (timatron) [inline tests/test-photonfill-unit-test-case.php]: check for an error?

File: tests/test-photonfill-unit-test-case.php [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		wp_update_attachment_metadata( $id, wp_generate_attachment_metadata( $id, $upload['file'] ) );
+		return $id;
+	}
+
+	/**
+	 * Activate My Photon for testing
+	 */
+	private function activate_my_photon() {
+		$my_photon_settings = array( 
+			'base-url'  => 'http://cdn.alley.dev/',
```
Comment (timatron) [inline tests/test-photonfill-unit-test-case.php]: put this in the documentation, that the phpunit tests will only run on a local alley machine

**Comment by davisshaver** [comment]
Most of the PHPCS issues have been cleaned up for `WordPress`, `WordPress-VIP`, and `WordPress-Docs`.

A few in Photonfill main class that might need ignores or patches...

```
FILE: ...Users/Davis/broadway/plugins/photonfill/php/class-photonfill.php
----------------------------------------------------------------------
FOUND 5 ERRORS AND 2 WARNINGS AFFECTING 6 LINES
----------------------------------------------------------------------
  204 | ERROR   | Intermediate images do not exist on the VIP
      |         | platform, and thus get_intermediate_image_sizes()
      |         | returns an empty array() on the platform. This
      |         | behavior is intentional to prevent WordPress from
      |         | generating multiple thumbnails when images are
      |         | uploaded.
  527 | ERROR   | Detected usage of a non-validated input variable:
      |         | $_POST
  527 | ERROR   | Processing form data without nonce verification.
  597 | WARNING | Detected access of super global var $_POST,
      |         | probably needs manual inspection.
  598 | WARNING | Detected access of super global var $_POST,
      |         | probably needs manual inspection.
```

**Comment by davisshaver** [comment]
@willgladstone, looking at the PHPCS issues above. In https://github.com/alleyinteractive/photonfill/blob/b5f6164e95c948ed848c2c4121e585e52fd4b60c/php/class-photonfill.php#L527, just wondering what lead to the `$_POST` check here? Maybe we can do it another way rather than adding nonce.

File: tests/test-photonfill-unit-test-case.php (hunk 3 above)
Comment (davisshaver) [inline tests/test-photonfill-unit-test-case.php]: We can just fail early in the test

**Comment by willgladstone** [comment]
Well, The intermediate images thing is something I already disabled as well. So it's irrelevant here. Not sure the best way to make it not complain. In terms of the POST check, that is me dealing with what core is giving us. There might be a nonce in there I can scrounge up. Let me look.

**Pull request merged**